### PR TITLE
Deprecate FID and remove other deprecated APIs

### PR DESCRIPTION
**Pull request by tunetheweb**
Makes the following changes assumiong the next release after this merge is a major version bump:

- Removes old deprecated APIs (`getXXX` calls and `ReportCallback`)
- Marks `onFID` as deprecated and to be removed in next major version
- Removes polyfill base builds as per #238 (note this includes `NavigationTimingPolyfillEntry`)
- Updates all examples in README from `onFID` to `onINP`
- Updates INP to a CWV in README
- Notes FID as deprecated in README

Notable it does NOT:
- Remove the FID polyfill. It's need to emit a FID on bfcache restore as browser doesn't 😔
- Console.warn on onFID use. As we're in a handover stage I think it's a little early for that. We can add that in a few months without a major release IMHO.

### CODE REVIEW

**Comment by tunetheweb** [comment]
Test all now passing, but leaving this in draft until we're ready to merge, nearer 12th March. However I'm not planning on adding more to this PR so it is ready for review in the meantime.

File: README.md [hunk 1]
```diff
@@ -466,7 +418,7 @@ _**Note:** see [the Page Lifecycle guide](https://developers.google.com/web/upda
 
 ## Build options
 
-The `web-vitals` package includes builds for the "standard", "attribution", and "base+polyfill" ([deprecated](https://github.com/GoogleChrome/web-vitals/issues/238)) builds, as well as different formats of each to allow developers to choose the format that best meets their needs or integrates with their architecture.
+The `web-vitals` package includes builds for the "standard" and "attribution" builds, as well as different formats of each to allow developers to choose the format that best meets their needs or integrates with their architecture.
```
Comment (philipwalton) [inline README.md:421]:
```suggestion
The `web-vitals` package includes both "standard" and "attribution" builds, as well as different formats of each to allow developers to choose the format that best meets their needs or integrates with their architecture.
```

**Review by philipwalton — CHANGES_REQUESTED** [review]
Overall LGTM. I left a few suggestions in this PR. In addition, can you:

- Create a `v4` branch and update this PR's merge target to that (instead of `main`).
- Remove references to the "base" script still in `package.json`  (in "exports" and "files").

File: src/onFID.ts [hunk 2]
```diff
@@ -79,23 +80,16 @@ export const onFID = (onReport: FIDReportCallback, opts?: ReportOpts) => {
     if (po) {
       onHidden(
         runOnce(() => {
-          handleEntries(po.takeRecords() as FIDMetric['entries']);
-          po.disconnect();
+          if (po) {
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/onFID.ts:83]: Ah I backed out some other change so you're right this isn't needed anymore.

File: src/onFID.ts [hunk 3]
```diff
@@ -105,27 +99,10 @@ export const onFID = (onReport: FIDReportCallback, opts?: ReportOpts) => {
           opts!.reportAllChanges,
         );
 
-        window.webVitals.resetFirstInputPolyfill();
-        window.webVitals.firstInputPolyfill(
-          handleEntry as FirstInputPolyfillCallback,
-        );
+        // Browsers don't re-emit FID on bfcache restore so fake it until you make it
+        resetFirstInputPolyfill();
+        -firstInputPolyfill(handleEntry as FirstInputPolyfillCallback);
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/onFID.ts:104]: Nope. From when I put back from a `git diff` 😁

**Review by philipwalton — CHANGES_REQUESTED** [review]
Still need to remove references to the "base" script in package.json (in "exports" and "files").

**Comment by tunetheweb** [comment]
Done!

**Review by philipwalton — APPROVED** [review]

**Pull request merged**